cpplint_wrapper_test fails on Windows

Starting with a fresh clone of drake I did a virgin Debug build on Windows (64 bit). Superbuild was clean in Visual Studio. I switched to build/drake/drake.sln; build there was clean (and fast since everything was already built). I ran the test suite and everything worked except `cpplint_wrapper_test`. 

I do have cpplint in my path and it appears that the test ran but failed. I didn't see it in the list of tests or in the bin directory so I'm not sure how to get more information about the failure.

Some not-very-helpful output:
```
1>        Start 14: cpplint_wrapper_test
1>  14/87 Test #14: cpplint_wrapper_test .....................***Failed    4.02 sec
The following tests FAILED:
1>  	 14 - cpplint_wrapper_test (Failed)
1>  Errors while running CTest
1>C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(133,5):
error MSB3073: The command "setlocal ... VCEnd" exited with code 8.
```

Below is actual output. Does this speak to you at all @jwnimmer-tri?
```
C:\Users\Sherm\Documents\GitHub\sherm1>python drake-distro/drake/common/test/cpplint_wrapper_test.py
.FF.FF
======================================================================
FAIL: test_huge (__main__.TestStringMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 104, in test_huge
    r"TOTAL 25 files passed"])
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 37, in run_and_expect
    self.assertRegexpMatches(output, item)
AssertionError: Regexp didn't match: 'Checking \\.+\\n' not found in 'Checking ............................\r\n TOTAL 25 files passed\r\n'

======================================================================
FAIL: test_ignored_extension (__main__.TestStringMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 82, in test_ignored_extension
    r"TOTAL 0 files"])
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 37, in run_and_expect
    self.assertRegexpMatches(output, item)
AssertionError: Regexp didn't match: 'Ignoring .*/cpplint_wrapper_test.py' not found in 'Ignoring C:\\Users\\Sherm\\Documents\\GitHub\\sherm1\\drake-distro\\drake\\common\\test\\cpplint_wrapper_test.py; not a valid file name.\r\nChecking ...\r\n TOTAL 0 files passed\r\n'

======================================================================
FAIL: test_summarize (__main__.TestStringMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 51, in test_summarize
    r"TOTAL 1 files passed"])
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 37, in run_and_expect
    self.assertRegexpMatches(output, item)
AssertionError: Regexp didn't match: 'Checking \\.+\\n' not found in 'Checking ....\r\n TOTAL 1 files passed\r\n'

======================================================================
FAIL: test_true_positive (__main__.TestStringMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 67, in test_true_positive
    r"TOTAL 1 files passed"])
  File "drake-distro/drake/common/test/cpplint_wrapper_test.py", line 37, in run_and_expect
    self.assertRegexpMatches(output, item)
AssertionError: Regexp didn't match: 'Checking \\.+\\n' not found in 'Checking ....\r\n TOTAL 1 files passed\r\n'

----------------------------------------------------------------------
Ran 6 tests in 3.065s

FAILED (failures=4)
```

```
AssertionError: Regexp didn't match:
  'Checking \\.+\\n'
not found in
  'Checking ............................\r\n TOTAL 25 files passed\r\n'
```

DOS newlines for the win!  I wonder why CI isn't complaining.

--- a/drake/common/test/cpplint_wrapper_test.py
+++ b/drake/common/test/cpplint_wrapper_test.py
@@ -47,7 +47,7 @@ class TestStringMethods(unittest.TestCase):
         self.run_and_expect(
             [filename],
             0,
-            [r"Checking \.+\n",
+            [r"Checking \.+\r?\n",
              r"TOTAL 1 files passed"])
         self.run_and_expect(
             ["--no-summarize", filename],
@@ -63,13 +63,13 @@ class TestStringMethods(unittest.TestCase):
         self.run_and_expect(
             [filename],
             0,
-            [r"Checking \.+\n",
+            [r"Checking \.+\r?\n",
              r"TOTAL 1 files passed"])
         # Test that pass-through arguments works.
         self.run_and_expect(
             [filename, "--", "--linelength=40"],
             1,
-            [r"Checking \.+\n",
+            [r"Checking \.+\r?\n",
              r"TOTAL 1 files checked",
              r"found [123] warnings",
              r"whitespace/line_length"])
@@ -78,7 +78,7 @@ class TestStringMethods(unittest.TestCase):
         self.run_and_expect(
             [self.get_absolute_filename('cpplint_wrapper_test.py')],
             0,
-            [r"Ignoring .*/cpplint_wrapper_test.py",
+            [r"Ignoring .*[/\\]cpplint_wrapper_test.py",
              r"TOTAL 0 files"])
 
     def test_num_processes(self):
@@ -100,7 +100,7 @@ class TestStringMethods(unittest.TestCase):
         self.run_and_expect(
             ["--num-processes=2"] + [filename] * 25,
             0,
-            [r"Checking \.+\n",
+            [r"Checking \.+\r?\n",
              r"TOTAL 25 files passed"])
 
 

